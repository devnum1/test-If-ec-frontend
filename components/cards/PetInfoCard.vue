// 01EZTX9M3AWK0ZB16B6KYR56F1

<template>
  <div class="pet-info-card">
    <div class="pet-info info-title">
      <div class="pet-info__name">“{{ information.name }}”</div>
      <div class="">
        {{ $t('common.tell_some_pet_info') }}
      </div>
    </div>
    <div class="pet-info info-row">
      <div class="info-label">
        {{ $t('common.gender.gender_title') }}
      </div>
      <div class="info-description">
        {{
          parseInt(information.gender) == 0
            ? $t('common.gender.boy')
            : $t('common.gender.girl')
        }}
      </div>
    </div>
    <div class="pet-info info-row">
      <div class="info-label">
        {{ $t('common.contraception_status.title') }}
      </div>
      <div class="info-description">
        {{ information.contraception }}
      </div>
    </div>
    <div class="pet-info info-row">
      <div class="info-label">
        {{ $t('common.dob.title') }}
      </div>
      <div class="info-description">
        {{ information.dob }}
      </div>
    </div>
  </div>
</template>
<script type="text/javascript">
export default {
  props: {
    petInfo: {
      type: Object,
      default: () => ({}),
    },
  },
  data() {
    return {
      progress: 1,
      information: {},
    };
  },
  updated() {},
  mounted() {
    this.petInfo ? (this.information = this.petInfo) : {};
  },
  methods: {},
};
</script>

<style lang="scss" scoped>
.pet-info-card {
  border: 1px solid rgba($green-border, 0.15);
  border-radius: 6px;
  text-align: center;
  background-color: white;
  display: flex;
  flex-direction: column;

  .pet-info {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 13px 20px;

    &:not(:first-child) {
      border-top: 1px dashed rgba($green-border, 0.15);
    }
  }

  .info {
    &-title {
      display: block;
      font-size: 18px;
      padding: 13px 13px;

      div {
        display: inline;
        color: rgba($black, 0.6);

        &.pet-info__name {
          color: $red;
        }
      }
    }

    &-label {
      font-size: 14px;
      color: rgba($black, 0.6);
    }
  }
}
</style>
