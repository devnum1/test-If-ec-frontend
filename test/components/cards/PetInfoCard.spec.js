import { shallowMount, config } from '@vue/test-utils';
import PetInfoCard from '@/components/cards/PetInfoCard.vue';

config.mocks.$t = (key) => key;

describe('PetInfoCard', () => {
  const petInfo = {
    name: 'うめじまのうめ子',
    dob: '2018.01.01',
    type: 'スタッフォードシャーブルテリア',
    contraception: 'なし',
    contract: '70％補償プラン',
    image: '',
    species: 0, // 1 => いぬ, 0 => ねこ
    gender: 1, // 1 => 男の子, 0 => 女の子
  };

  const flushPromises = () => new Promise(setImmediate);
  let wrapper;

  const factory = (computed = {}) => {
    return shallowMount(PetInfoCard, {
      propsData: {
        petInfo: petInfo,
      },
    });
  };

  beforeEach(() => {
    wrapper = factory();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  test('is a Vue instance', () => {
    const wrapper = shallowMount(PetInfoCard);
    expect(wrapper.vm).toBeTruthy();
  });

  test('match Snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('Pet Info card to display info', async () => {
    await flushPromises();

    expect(wrapper.find('.pet-info__name').exists()).toBe(true);
    const name = wrapper.find('.pet-info__name');
    expect(name.text()).toBe('“' + petInfo.name + '”');

    expect(wrapper.find('.pet-info.info-row').exists()).toBe(true);

    expect(wrapper.findAll('.info-description').at(1).text()).toBe(
      petInfo.contraception
    );
  });
});
