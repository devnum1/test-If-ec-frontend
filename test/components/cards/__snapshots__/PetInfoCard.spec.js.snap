// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PetInfoCard match Snapshot 1`] = `
"<div class=\\"pet-info-card\\">
  <div class=\\"pet-info info-title\\">
    <div class=\\"pet-info__name\\">
      “うめじまのうめ子”
    </div>
    <div>
      common.tell_some_pet_info
    </div>
  </div>
  <div class=\\"pet-info info-row\\">
    <div class=\\"info-label\\">
      common.gender.gender_title
    </div>
    <div class=\\"info-description\\">
      common.gender.girl
    </div>
  </div>
  <div class=\\"pet-info info-row\\">
    <div class=\\"info-label\\">
      common.contraception_status.title
    </div>
    <div class=\\"info-description\\">
      なし
    </div>
  </div>
  <div class=\\"pet-info info-row\\">
    <div class=\\"info-label\\">
      common.dob.title
    </div>
    <div class=\\"info-description\\">
      2018.01.01
    </div>
  </div>
</div>"
`;
